### PR TITLE
feat: treat ambiguous typed-link targets as broken in find_broken_links

### LINKED ISSUE #55: feat: treat ambiguous typed-link targets as broken in find_broken_links
## Problem statement

`find_broken_links` currently treats a typed-link target (`to_target`) as broken only when it resolves to **zero** notes.

When `to_target` resolves to **multiple** notes (duplicate titles / note IDs / path matches), the link is ambiguous and effectively unusable for a deterministic typed-link graph, but it is not reported.

Example:

- Vault contains both `Projects/Project A.md` and `Archive/Project A.md`
- A note contains `part_of: [[Project A]]`
- The resolver returns 2+ paths, but `find_broken_links` reports nothing today.

## Proposed solution

Add an option to `find_broken_links`:

- `treat_ambiguous_as_broken` (boolean, default: `true`)

Resolution rule:

- `resolved.length === 0` → broken (existing behavior)
- `resolved.length === 1` → OK (existing behavior)
- `resolved.length >= 2` → broken when `treat_ambiguous_as_broken=true`
  - Include the candidate matches in the existing `resolutions[]` field so users can disambiguate.
  - Cap `resolutions[]` by `max_resolutions_per_target` (keep default `5`).

Notes:

- `broken_total` should include both unresolved (0) and ambiguous (2+) targets when `treat_ambiguous_as_broken=true`.

Acceptance criteria:

- Ambiguous targets (2+) appear in `broken[]` with `resolutions.length >= 2`.
- Missing targets (0) still appear in `broken[]` with `resolutions=[]`.
- Tool stays DB-only and respects safety bounds (`max_links`, `max_broken`, `max_resolutions_per_target`).

## Constraints / context (optional)

- Component: MCP server (`packages/mcp`)
- Update docs and HTTP tests accordingly.
- Related: #24 stopped indexing body wikilinks; typed-link hygiene is now more important.

### PR DESCRIPTION
**Pull request by maehwasoo**
## What

- Add `treat_ambiguous_as_broken` to `find_broken_links` (default `true`).
- Treat ambiguous targets (2+ resolutions) as broken and return candidate `resolutions[]` (capped by `max_resolutions_per_target`).
- Add an HTTP test covering ambiguous targets vs flag disabled.

## Why

- Make typed-link hygiene actionable by surfacing ambiguous targets during audits.
- Fixes #55

## How

- Extend tool input schema and broken-detection logic in `packages/mcp/src/tools/findBrokenLinks.ts`.
- Add a synthetic DB fixture with two matching notes and assert returned `resolutions[]`.
- Update `docs/reference/mcp-tools.md`.
- Testing: `pnpm build`, `pnpm test`

### CODE REVIEW

File: packages/mcp/src/tools/findBrokenLinks.ts [hunk 1]
```diff
@@ -158,7 +162,10 @@ export function registerFindBrokenLinksTool(server: McpServer, deps: McpToolDeps
         if (!target) continue;
 
         const resolved = resolveCached(target);
-        if (resolved.length > 0) continue;
+        const isUnresolved = resolved.length === 0;
+        const isAmbiguous = resolved.length >= 2;
+        const isBroken = isUnresolved || (args.treat_ambiguous_as_broken && isAmbiguous);
```
[the remark this answers is not in the document]
Comment (maehwasoo) [reply packages/mcp/src/tools/findBrokenLinks.ts:167]: @codex 
Added a regression test for the max_resolutions_per_target=1 case as well. Let me know if you see any other review items or edge cases I should cover.
